src/golang/lib/job/lambda.go

```diff
@@ -68,18 +156,34 @@ func (j *lambdaJobManager) Launch(ctx context.Context, name string, spec Spec) e
 		return errors.Wrap(err, "Unable to marshal request payload.")
 	}
 
-	functionName, err := mapJobTypeToLambdaFunction(spec)
-	if err != nil {
-		return errors.Wrap(err, "Unable to launch job.")
+	// Lambda functions with custom memory configurations should be executed synchronously.
+	// This is to prevent such memory configurations from bleeding out to other operators using
+	// the same lambda function. We reset the memory configuration at the very end of this function.
+	// NOTE: this does not provide perfect isolation. It is still possible for operators scheduled
+	// before to race with the memory configuration update, since regular lambda functions are run
+	// asynchronously, and we have no visibility into the AWS's event queue.
+	invocationType := aws.String("Event")
+	if previousMemoryMB != nil {
+		invocationType = aws.String("RequestResponse")
 	}
 
 	invokeInput := &lambda.InvokeInput{
 		FunctionName:   &functionName,
-		InvocationType: aws.String("Event"),
+		InvocationType: invocationType,
 		Payload:        payload,
 	}
 
-	_, err = j.lambdaService.Invoke(invokeInput)
+	// Resetting memory back to its original value is best-effort.
```

Comment:
Should we define the deferred function before the update call? That way if the update fails we can set this back to the original memory setting.